test/tests/smoke_test.rs

```diff
@@ -30,7 +31,13 @@ fn smoke_test() {
     })
     .unwrap();
     let mut output = vec![];
-    hw.copy_output_until_exit_success(&mut output).unwrap();
+
+    // Step until runtime firmware denotes successful firmware upload
+    hw.step_until(|m| m.soc_mbox().status().read().status().cmd_complete());
```

Comment:
Does this actually do what you think it does? hw.upload_firmware() already waits for the upload command to complete:

https://github.com/chipsalliance/caliptra-sw/blob/effcd075176fdb598ba103a43e7aa99ac9f06b09/hw-model/src/lib.rs#L385-L399

I don't think the strings we are looking for below will be found.

Reply:
Discussed offline. Switched to checking for UART output from RT.